1.process_data/nbconverted/single_cell_qc.py

```diff
@@ -0,0 +1,277 @@
+            "Image_Metadata_Site",
+        ]
+    ].to_dict(orient="records")
+}
+# fmt: on
+
+next(iter(outline_to_orig_mapping.items()))
+
+
+# ## Detect technically mis-segmented nuclei
+# 
+# We define technically mis-segmented nuclei as segmentations that include more than one nuclei, under or over-segmentation, or segmentation of background/smudging.
+# 
+# We are using the measurements in two different conditions:
+# 
+
+# 1. Irregular shaped nuclei where the intensity based center is much different than the shape's center
+# 
+# This condition looks to detect any technical outliers with irregular shape and very high difference in centeroids, which can be detecting multiple different types of technical outliers.
+# 
+# - `FormFactor`, which detects how irregular the shape is. 
+# - `MassDisplacement`, which detects how different the segmentation versus intensity based centeroids are (which can reflect multiple nuclei within one segmentation).
+# We understand that interesting phenotypes will occur, so we are going to evaluate if this will identify the mis-segmentationsb
+
+# ### Detect outliers and show single-cell image crops with CytoDataFrame (cdf)
+
+# In[6]:
+
+
+# find irregular shaped nuclei using thresholds that maximizes
+# removing most technical outliers and minimizes good cells
+feature_thresholds = {
+    "Nuclei_AreaShape_FormFactor": -2.35,
+    "Nuclei_Intensity_MassDisplacement_DNA": 1.5
+}
+
+irregular_nuclei_outliers = cosmicqc.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds=feature_thresholds
+)
+
+irregular_nuclei_outliers_cdf = CytoDataFrame(
+    data=pd.DataFrame(irregular_nuclei_outliers),
+    data_context_dir=f"/media/jenna/8TB-C/work/JUMP-download-images/JUMP-single-cell/0.download_data/data/images/{plate_id}/orig",
+    data_outline_context_dir=f"/media/jenna/8TB-C/work/JUMP-download-images/JUMP-single-cell/0.download_data/data/images/{plate_id}/outlines",
+    segmentation_file_regex=outline_to_orig_mapping,
+)[
+    [
+        "Nuclei_AreaShape_FormFactor",
+        "Nuclei_Intensity_MassDisplacement_DNA",
+        "Image_FileName_OrigDNA",
+    ]
+]
+
+
+print(irregular_nuclei_outliers_cdf.shape)
+irregular_nuclei_outliers_cdf.sort_values(by="Nuclei_AreaShape_FormFactor", ascending=False).head(2)
+
+
+# ### Randomly sample outlier rows to visually inspect if the threshold looks to be optimized
+
+# In[7]:
+
+
+irregular_nuclei_outliers_cdf.sample(n=5, random_state=0)
+
+
+# ### Generate a new dataframe to save the original indices for filtering prior to further preprocessing
+
+# In[8]:
+
+
+# Create a new dataframe for failed QC indices
+failed_qc_indices = irregular_nuclei_outliers[metadata_cols].copy()
+failed_qc_indices["original_index"] = failed_qc_indices.index  # Store original index
+failed_qc_indices = failed_qc_indices.reset_index(drop=True)  # Reset index
+failed_qc_indices["cqc.formfactor_displacement_outlier"] = True
```

Comment:
Minor suggestion for clarity about the boolean value indicator (this also follows the `cosmicqc` convention).
```suggestion
failed_qc_indices["cqc.formfactor_displacement_outlier.is_outlier"] = True
```

Reply:
Done and done!